### PR TITLE
SOLR-16746: Migrate How to Contribute Content from cwiki

### PR DESCRIPTION
**Pull request by epugh**
https://issues.apache.org/jira/browse/SOLR-16746

# Description

Content is hidden in Confluence wiki, is out of date, and hard to update.

# Solution

Migrate content

# Tests
manual

# Checklist

Please review the following and check all that apply:

- [ ] I have reviewed the guidelines for [How to Contribute](https://wiki.apache.org/solr/HowToContribute) and my code conforms to the standards described there to the best of my ability.
- [ ] I have created a Jira issue and added the issue ID to my pull request title.
- [ ] I have given Solr maintainers [access](https://help.github.com/en/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork) to contribute to my PR branch. (optional but recommended)
- [ ] I have developed this patch against the `main` branch.
- [ ] I have run `./gradlew check`.
- [ ] I have added tests for my changes.
- [ ] I have added documentation for the [Reference Guide](https://github.com/apache/solr/tree/main/solr/solr-ref-guide)

### CODE REVIEW

**Comment by HoustonPutman** [comment]
Ok I made a ton of changes to the git page, better reflecting best practices and using asciidoc stuff instead of markdown.

File: dev-docs/git.adoc [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+
+Once the conflicts have been resolved:
+
+[console]
+----
+$ git commit # Commit the fixes for conflicts. The message may be already populated, edit it as needed but it's not very important.
+$ git push fork branch # Push the merged feature branch to your fork
+----
+
+Git may prompt you to `git rebase --continue`, if conflicts are resolved, that should end up in the same place, of prompting you to git push your final changes to the remote repository.
```
Comment (HoustonPutman) [inline dev-docs/git.adoc:122]: After my changes, I don't think this is the case. There shouldn't be any rebase command used...

File: dev-docs/git.adoc [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+
+You may want to consider doing `git merge --squash main` - this will merge all commits from `main` into a single commit locally.
```
Comment (HoustonPutman) [inline dev-docs/git.adoc:124]: I'm not sure this is really necessary either. Github will filter out all of the merged commits in the PR view.

File: dev-docs/git.adoc (hunk 1 above)
Comment (epugh) [reply dev-docs/git.adoc:122]: I think we should remove this...   It's easier to add new docs versus get rid of them...

**Comment by HoustonPutman** [comment]
Also can we make a folder for these called `dev-docs/contributing`. Much better than having a lot of random docs mixed together.

**Comment by epugh** [comment]
G

> Also can we make a folder for these called `dev-docs/contributing`. Much better than having a lot of random docs mixed together.

I'm excited that we are building enough docs in `dev-docs` to start thinking about nesting...  Having said that, I don't think we actually have enough for a multiples in in a `dev-docs/contributing`?   What structure are you thinking would go in there.....?

**Comment by HoustonPutman** [comment]
Yeah maybe there aren't enough files yet, but something to think about in the future!

**Review by janhoy — APPROVED** [review]

**Pull request merged**